### PR TITLE
CTPPS conditions database layout

### PR DESCRIPTION
**Pull request by malbouis**
code to account for CTPPS Conditions Database objects in CondCore/Utilities and CondCore/CTPPSPlugins

CalibForward package that contains analyzers to write sqlite file and retrieve conditions DB objects.

### CODE REVIEW

**Comment by slava77** [comment]
assign alca

**Comment by slava77** [comment]
[following up after a private email from @fabferro ]
by just name association, it looks like CalibForward/CTPPSPixelCalibration belongs to alca.
So, I added a signature.
I'm expecting that the new packages (CalibForward/CTPPSPixelCalibration CondCore/CTPPSPlugins) are assigned to a relevant category after review by corresponding area reviewers and 
a PR for https://github.com/cms-sw/cms-bot/blob/master/categories.py is then made to assign the category

**Comment by arunhep** [comment]
please test

**Comment by slava77** [comment]
@cmsbuild please test

``` Caught a CMSSW exception when running connection recovery. ```
hopefully this is transient.

**Comment by franzoni** [comment]
+1

**Comment by franzoni** [comment]
@malbouis

FYI
etrieve-sqlite-daqmap-analysismask_cfg.py and
write-ctpps-pixel-analysismask_cfg.py
are good tools to have to produce sqlite from release-bound content.

For when you wanted  to create an sqlite for payloads already existing in database,
there's no need to use CTPPSPixelDAQMappingAnalyzer 
and WriteCTPPSPixelAnalysisMask
rather you can use tools available to all detector/alcadb 
see instructions here https://indico.cern.ch/event/507993/contributions/2020446/ slide 3 onwards

**Comment by ggovi** [comment]
+1

File: CalibForward/CTPPSPixelCalibration/BuildFile.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<use   name="CondFormats/CTPPSReadoutObjects"/>
```
Comment (davidlange6) [inline CalibForward/CTPPSPixelCalibration/BuildFile.xml]: hi @malbouis - it looks like the right subsystem for this package is CondTools. (eg, CondTools/CTPPS to keep things general)

File: CalibForward/CTPPSPixelCalibration/BuildFile.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+<use   name="CondCore/DBOutputService"/>
+<use   name="FWCore/ServiceRegistry"/>
+<use   name="FWCore/Framework"/>
+<use name="CondFormats/DataRecord"/>
+<use name="DataFormats/CTPPSDetId"/>
+  <use   name="root"/>
```
Comment (davidlange6) [inline CalibForward/CTPPSPixelCalibration/BuildFile.xml]: this should not be needed?

File: CalibForward/CTPPSPixelCalibration/plugins/CTPPSPixelDAQMappingAnalyzer.cc [hunk 3]
```diff
@@ -0,0 +1,82 @@
+#include <string>
+#include <iostream>
+#include <map>
```
Comment (davidlange6) [inline CalibForward/CTPPSPixelCalibration/plugins/CTPPSPixelDAQMappingAnalyzer.cc]: is this include needed?

File: CalibForward/CTPPSPixelCalibration/plugins/CTPPSPixelDAQMappingAnalyzer.cc [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelDAQMapping.h"
+#include "CondFormats/DataRecord/interface/CTPPSPixelDAQMappingRcd.h"
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelAnalysisMask.h"
+#include "CondFormats/DataRecord/interface/CTPPSPixelAnalysisMaskRcd.h"
+
+using namespace std;
+
+namespace edmtest
+{
+  class CTPPSPixelDAQMappingAnalyzer : public edm::EDAnalyzer
```
Comment (davidlange6) [inline CalibForward/CTPPSPixelCalibration/plugins/CTPPSPixelDAQMappingAnalyzer.cc]: please use a non-legacy module type

File: CalibForward/CTPPSPixelCalibration/BuildFile.xml (hunk 1 above)
Comment (malbouis) [inline CalibForward/CTPPSPixelCalibration/BuildFile.xml]: I moved it to CondTools/CTPPS

File: CalibForward/CTPPSPixelCalibration/BuildFile.xml (hunk 2 above)
Comment (malbouis) [inline CalibForward/CTPPSPixelCalibration/BuildFile.xml]: CondFormats/DataRecord is needed for the data records, the others are not. I removed them.

File: CalibForward/CTPPSPixelCalibration/plugins/CTPPSPixelDAQMappingAnalyzer.cc (hunk 3 above)
Comment (malbouis) [inline CalibForward/CTPPSPixelCalibration/plugins/CTPPSPixelDAQMappingAnalyzer.cc]: I removed it now.

File: CalibForward/CTPPSPixelCalibration/plugins/CTPPSPixelDAQMappingAnalyzer.cc (hunk 4 above)
Comment (malbouis) [inline CalibForward/CTPPSPixelCalibration/plugins/CTPPSPixelDAQMappingAnalyzer.cc]: I removed the legacy module type.

**Comment by arunhep** [comment]
please test

**Comment by ggovi** [comment]
+1

**Comment by franzoni** [comment]
+1

**Pull request merged**